### DISCUSSION TITLE
Request Memoization and Full Route Cache Limits

### DISCUSSION THREAD

**Discussion by xyera2** [vercel/next.js#64667] [App Router]
**Context:**  I've got a Next.js app with several `page.tsx` files (all located within a `[lang]` directory). Each `page.tsx` file handles data fetching from various APIs with `fetch` ( `lang` param is passed to each API call). I want to refresh the data from these APIs on every page every 2 minutes. Also, I want to render these pages on the server every 2 minutes so they could be cached on CDN (to reduce the load on origin). That is why I'm exporting these variables from each page.tsx file:

```js
export const dynamic = 'force-static';
export const revalidate = 120;
```

**Problem**: This setup has worked fine so far for me, but lately I started fetching a LOT of data from those APIs on some pages. I've been getting a lot of `failed to set fetch cache... fetch for over 2MB of data can not be fetched` errors in dev (`next dev`) mode because of that. However, everything seems to be working fine in prod (`next start`) and full route seems to be correctly cached.

**Questions**:
1.  What exactly are the limits of the fetch cache and the full route cache? In dev mode, can I just disregard "failed to set fetch cache" errors if the full route cache correctly caches the server-side rendering result? 
2. Is there a way to enable the fetch cache in dev mode and increase the fetch cache size? Given that I'm fetching a lot of data from these APIs, it's causing a noticeable delay when navigating to some pages in dev mode, as the data is fetched each time and the revalidate value is essentially ignored? (I found this similar [discussion](https://github.com/vercel/next.js/discussions/48481), but still not sure what is the best way to improve dev experience.)
3. Is exporting those specific values from my `page.tsx` file the best approach for my use case? I exported those values because introducing the `[lang]` directory increased the load on my origin server due to server-side re-renders for every request.

Thanks in forward to anyone who responds and tries to help!*

**Comment by joepetrillo** [comment]
Also running into this - any updates?
